amqp-postgres/amqp_postgres/amqp_consumer.py

```diff
@@ -0,0 +1,62 @@
+########
+# Copyright (c) 2018 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+############
+
+import json
+from uuid import uuid4
+
+from flask import current_app
+
+
+class AMQPLogsEventsConsumer(object):
+    LOGS_EXCHANGE = 'cloudify-logs'
+    EVENTS_EXCHANGE = 'cloudify-events'
+
+    def __init__(self, message_processor):
+        self.queue = str(uuid4())
+        self._connection = None
+        self._in_channel = None
+        self._message_processor = message_processor
+
+        # This is here because AMQPConnection expects it
+        self.routing_key = ''
+
+    def register(self, connection):
+        self._connection = connection
+        self._in_channel = connection.channel()
+        self._register_queue(self._in_channel)
+
+    def _register_queue(self, channel):
```

Comment:
might as well combine the two methods, this and `register`, the split i think does nothing